common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java

```diff
@@ -92,26 +92,6 @@ public static CalendarInterval fromString(String s) {
     }
   }
 
-  /**
-   * Convert a string to CalendarInterval. Unlike fromString, this method is case-insensitive and
-   * will throw IllegalArgumentException when the input string is not a valid interval.
-   *
-   * @throws IllegalArgumentException if the string is not a valid internal.
-   */
-  public static CalendarInterval fromCaseInsensitiveString(String s) {
-    if (s == null || s.trim().isEmpty()) {
-      throw new IllegalArgumentException("Interval cannot be null or blank.");
-    }
-    String sInLowerCase = s.trim().toLowerCase(Locale.ROOT);
```

Comment:
Previously, it seems that we support `trim` additionally. I'm wondering if `Pattern.CASE_INSENSITIVE` covers that, too.

Reply:
`fromString` trims before pattern matching as well: https://github.com/apache/spark/blob/79e204770300dab4a669b9f8e2421ef905236e7b/common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java#L78